### PR TITLE
perf(boost): Favorite artists perf

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Boosts performance of Favorite artists surface by:

- converting to func component
- using flashlist over flatlist
- memoizing the list
- useCallback for listItem and keyExtractor

|Before|After|
|---|---|
|<video src="https://github.com/user-attachments/assets/8ebf4ca8-0d59-4a4a-b48a-1b33c8a4e469" width={400} />|<video src="https://github.com/user-attachments/assets/eb3885d1-234d-4a73-ab89-0595935a75ab" width={400} />|

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- boost Favorite artists perf - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (boost Favorite artists perf - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c4d137ed8a7200ae3e151b2ddf39a3521c1425c8
</p>

**Review by MounirDhahri — DISMISSED** [review]
Nice! at this point I am starting to feel that we should add an eslint rule against using Flatlist unless necessary

File: src/app/Scenes/Favorites/FavoriteArtists.tsx [hunk 1]
```diff
@@ -9,113 +9,126 @@ import { PAGE_SIZE } from "app/Components/constants"
...
       }
-      this.setState({ refreshingFromPull: false })
+      setRefreshingFromPull(false)
     })
-  }
-
-  // @TODO: Implement test on this component https://artsyproduct.atlassian.net/browse/LD-563
-  render() {
-    const rows = extractNodes(this.props.me.followsAndSaves?.artists)
+  }, [relay])
```
Comment (MrSltun) [inline src/app/Scenes/Favorites/FavoriteArtists.tsx:51]: 👏👏👏👏

**Review by MrSltun — DISMISSED** [review]
Thank you!! ⚡⚡⚡ 
I just left a non-blocking question

File: src/app/Scenes/Favorites/FavoriteArtists.tsx [hunk 2]
```diff
@@ -9,113 +9,126 @@ import { PAGE_SIZE } from "app/Components/constants"
...
+      item,
+    }: {
+      item: {
+        readonly artist:
+          | {
+              readonly id: string
+              readonly " $fragmentSpreads": FragmentRefs<"ArtistListItem_artist">
+            }
+          | null
+          | undefined
```
[the remark this answers is not in the document]
Comment (gkartalis) [reply src/app/Scenes/Favorites/FavoriteArtists.tsx]: removed the types they can be inferred from the file

**Comment by damassi** [comment]
> eslint rule against using Flatlist unless necessary

Great idea @MounirDhahri - something like this? https://github.com/artsy/volt/blob/main/.eslintrc.js#L127-L129

**Pull request merged**